ConferencesApp/ConferenceTableViewController.swift

```diff
@@ -117,11 +130,107 @@ class ConferenceTableViewController: UITableViewController, MKMapViewDelegate {
         let venueLocation:CLLocationCoordinate2D = CLLocationCoordinate2DMake(latitude, longitude)
-                let theRegion:MKCoordinateRegion = MKCoordinateRegionMake(venueLocation, theSpan)
-
+        let theRegion:MKCoordinateRegion = MKCoordinateRegionMake(venueLocation, theSpan)
+        
         return theRegion
     }
-
+    
+    func setTapGestureOnDateImage() {
+        let tapGesture = UITapGestureRecognizer()
+        tapGesture.numberOfTapsRequired = 1
+        tapGesture.addTarget(self, action: "tapOnDateImageHandler:")
+        dateImage.addGestureRecognizer(tapGesture)
+    }
+    
+    func tapOnDateImageHandler(sender: UITapGestureRecognizer) {
+        let eventStore = EKEventStore()
+        
+        eventStore.requestAccessToEntityType(EKEntityType.Event, completion: {
+            granted, error in
+            if (granted) && (error == nil) {
+                if self.calendarEventID.isEmpty {
+                     dispatch_async(dispatch_get_main_queue()) {
+                        self.addCalendarEventtoStore(eventStore)
+                    }
+                    
+                } else { // Event Already present in calendar.
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.deleteEventFromStore(eventStore, eventID:self.calendarEventID)
+                    }
+                }
+            }
+        })
+        
+    }
+    
+    func addCalendarEventtoStore(eventStore:EKEventStore) {
+        let event:EKEvent = EKEvent(eventStore: eventStore)
+        event.title = self.conferenceName
+        event.startDate = self.startDate
+        event.endDate = self.endDate
+        event.location = self.locationLabel.text
+        event.allDay = true
+        // event.notes = "Note for conference"
+        
+        event.calendar = eventStore.defaultCalendarForNewEvents
+        
+        let alertController = UIAlertController(title: "Create Event.",
+            message: "Create Event in Calender ?", preferredStyle: .Alert)
+        
+        let createAction = UIAlertAction(title: "Create", style: .Default) { (action) -> Void in
+            do {
+                try eventStore.saveEvent(event, span: EKSpan.ThisEvent, commit: true)
+                self.calendarEventID = event.eventIdentifier
+                
+                self.conferenceDataStore?.updateCalendarEventIdentifier(self.conference,
+                    eventId:event.eventIdentifier)
+            } catch {
+                print("Failed to create an event in Calendar.")
+            }
+        }
+        
+        let cancelAction = UIAlertAction(title: "Cancel", style: .Default, handler:nil )
+        
+        alertController.addAction(createAction)
+        alertController.addAction(cancelAction)
+        
+        self.presentViewController(alertController, animated: true, completion: nil)
+    }
+    
+    func deleteEventFromStore( eventStore:EKEventStore, eventID: String) {
```

Comment:
Colon Violation: When specifying a type, always associate the colon with the identifier